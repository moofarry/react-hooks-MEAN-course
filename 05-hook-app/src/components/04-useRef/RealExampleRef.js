import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  const handleShowHide = () => {
    setShow(!show);
  };

  return (
    <div>
      <h1>RealExampleRef</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button className="btn btn-primary mt-5" onClick={handleShowHide}>
        Show /Hide
      </button>
    </div>
  );
};
