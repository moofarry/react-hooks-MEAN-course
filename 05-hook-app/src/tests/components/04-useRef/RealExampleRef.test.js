import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe("test in <RealExampleRef/>", () => {
  const wrapper = shallow(<RealExampleRef />);
  test("should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("should be show component <MultipleCustomHooks/>", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
