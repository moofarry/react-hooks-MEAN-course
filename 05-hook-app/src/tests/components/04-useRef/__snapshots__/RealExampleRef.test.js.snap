// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <RealExampleRef/> should show correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show /Hide
  </button>
</div>
`;
